com.unity.multiplayer.mlapi/Tests/Editor/NetworkManagerCustomMessageManagerTests.cs

```diff
@@ -0,0 +1,40 @@
+using System.Collections.Generic;
+using MLAPI.Configuration;
+using MLAPI.Messaging;
+using NUnit.Framework;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+
+namespace MLAPI.EditorTests
+{
+    public class NetworkManagerCustomMessageManagerTests
+    {
+        [Test]
+        public void CustomMessageManagerAssigned()
+        {
+            var gameObject = new GameObject(nameof(CustomMessageManagerAssigned));
+            var networkManager = gameObject.AddComponent<NetworkManager>();
+            var transport = gameObject.AddComponent<DummyTransport>();
+
+            // MLAPI sets this in validate
+            networkManager.NetworkConfig = new NetworkConfig()
+            {
+                // Set the current scene to prevent unexpected log messages which would trigger a failure
+                RegisteredScenes = new List<string>() {SceneManager.GetActiveScene().name}
+            };
+
+            // Set dummy transport that does nothing
+            networkManager.NetworkConfig.NetworkTransport = transport;
+
+            CustomMessagingManager preManager = networkManager.CustomMessagingManager;
+
+            // Start server to cause init
+            networkManager.StartServer();
+
+            Debug.Assert(preManager == null);
+            Debug.Assert(networkManager.CustomMessagingManager != null);
```

Comment:
Assuming you like the 'base class' idea above, then you could test all of the "NetworkManager-extended" classes in a loop here.